src/main/java/com/github/matidominati/medicalclinic/personInfo/PatientController.java

```diff
@@ -0,0 +1,54 @@
+package com.github.matidominati.medicalclinic.personInfo;
+
+import lombok.Data;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@Data
+@RestController
+@RequestMapping("/patients")
+public class PatientController {
+    private final List<Patient> patients;
+    @GetMapping
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+    @GetMapping("/{email}")
+    public Patient getPatient (@PathVariable String email) {
+        return checkIfPatientExist(email);
+    }
+    @PostMapping
+    public Patient addPatient(@RequestBody Patient patient) {
+        if (!patients.contains(patient) && patient.getEmail() != null) {
+            patients.add(patient);
+            return patient;
+        } else {
+            throw new RuntimeException("Pacjent już istnieje w bazie");
```

Comment:
Ten message nie jest prawdziwy. W sytuacji gdy ktos wyslal pacjenta z emailem, ktory jest null pojawi sie uzytkownikowi komunikat ze Pacjent juz istnieje w bazie danych. 
To co powinnismy zrobic to rozbic to na dwie walidacje 
Ja polecam zawsze sprawdzac rzeczy, ktore sa nieprawidlowe tzn:

    public Patient addPatient(@RequestBody Patient patient) {
        if (patient.getEmail() == null) {
        throw new IllegalArgumentException("Email cannot be null");
        }
        if (patients.contains(patient) {
        throw new IllegalArgumentException("Patient exists in db.")
        }
        patients.add(patient);
        return patient;
}